test/N3Parser-test.js

```diff
@@ -1433,6 +1433,353 @@ describe('Parser', () => {
+                  ['_:b2', 'f:son', 'ex:joe']));
+
+    it('should parse a ^ path in a list as object',
+      shouldParse(parser, '@prefix : <ex:>. @prefix fam: <f:>.' +
+                          '<l> <is> (<x> :joe^fam:son <y>).',
+                  ['l', 'is', '_:b0'],
+                  ['_:b0', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#first', 'x'],
+                  ['_:b0', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#rest',  '_:b1'],
+                  ['_:b1', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#first', '_:b2'],
+                  ['_:b1', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#rest',  '_:b3'],
+                  ['_:b3', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#first', 'y'],
+                  ['_:b3', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#rest', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#nil'],
+                  ['_:b2', 'f:son', 'ex:joe']));
+
+    it('should parse a formula as list item',
+        shouldParse(parser, '<a> <findAll> ( <b> { <b> a <type>. <b> <something> <foo> } <o> ).',
+        ['a', 'findAll', '_:b0'],
+        ['_:b0', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#first', 'b'],
+        ['_:b0', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#rest', '_:b2'],
+        ['_:b2', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#first', 'o'],
+        ['_:b2', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#rest', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#nil'],
+        ['b', 'something', 'foo', '_:b1'],
+        ['b', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type', 'type', '_:b1']
+    ));
+
+    it('should not parse an invalid ! path',
+      shouldNotParse(parser, '<a>!"invalid" ', 'Expected entity but got literal on line 1.'));
+
+    it('should not parse an invalid ^ path',
+      shouldNotParse(parser, '<a>^"invalid" ', 'Expected entity but got literal on line 1.'));
+
+    it('should parse literal as subject',
+        shouldParse(parser, '<a> <b> {1 <greaterThan> 0}.',
+            ['a', 'b', '_:b0'],
+            ['"1"^^http://www.w3.org/2001/XMLSchema#integer', 'greaterThan', '"0"^^http://www.w3.org/2001/XMLSchema#integer', '_:b0']
+        ));
+
+    it('should parse literals with datatype as subject',
+        shouldParse(parser, '<a> <b> {"a"^^<c> <greaterThan> "b"^^<c>}.',
+            ['a', 'b', '_:b0'],
+            ['"a"^^http://example.org/c', 'greaterThan', '"b"^^http://example.org/c', '_:b0']
+        ));
+
+    it('should parse literals with language as subject',
+        shouldParse(parser, '<a> <b> {"bonjour"@fr <sameAs> "hello"@en}.',
+            ['a', 'b', '_:b0'],
+            ['"bonjour"@fr', 'sameAs', '"hello"@en', '_:b0']
+        ));
+
+    it('should not parse RDF* in the subject position',
+      shouldNotParse(parser, '<<<a> <b> <c>>> <a> <b> .',
+        'Unexpected RDF* syntax on line 1.'));
+
+    it('should not parse RDF* in the object position',
+      shouldNotParse(parser, '<a> <b> <<<a> <b> <c>>>.',
+        'Unexpected RDF* syntax on line 1.'));
```

Comment:
Though `RDF*` and `RDF-star` are pronounced the same, they should not be used interchangeably. `RDF*` identifies an old, draft idea. `RDF-star` identifies the current work-in-progress which is expected to be standardized in the next couple of years (Community Group [report](https://www.w3.org/2021/12/rdf-star.html), Working Group [repositories](https://www.w3.org/groups/wg/rdf-star/tools)). There are MAJOR differences between these, and confusion among them should be avoided whenever possible. I *believe* the intent here is `RDF-star`, but I could be wrong, so please review the linked documents, and make appropriate adjustments.
```suggestion
    it('should not parse RDF-star in the subject position',
      shouldNotParse(parser, '<<<a> <b> <c>>> <a> <b> .',
        'Unexpected RDF-star syntax on line 1.'));

    it('should not parse RDF-star in the object position',
      shouldNotParse(parser, '<a> <b> <<<a> <b> <c>>>.',
        'Unexpected RDF-star syntax on line 1.'));
```

Reply:
This PR targets the main branch which currently implements something closer to `RDF*`. These references will be updated when merged with the changes in https://github.com/rdfjs/N3.js/pull/311 which implements `RDF-star` and uses such terminology in its error messaging.